virttest/utils_kernel_module.py

```diff
@@ -134,8 +139,13 @@ def restore(self):
 
         if self._loaded_config is not None:
             cmds = []
-            # TODO: Handle cases were module cannot be removed
-            cmds.append('modprobe -r --remove-dependencies %s' % self._module_name)
+            unload_cmd = ('modprobe -r --remove-dependencies %s'
+                          % self._module_name)
+            try:
+                process.run(unload_cmd + ' --dry-run')
+                cmds.append(unload_cmd)
+            except process.CmdError:
+                cmds.append('modprobe -r %s' % self._module_name)
```

Comment:
Just suggestion:Might we pick up them into a private function to avoid redundantly.

Reply:
Thanks, introduced a new method.